app/controllers/uploads_controller.rb

```diff
@@ -270,6 +301,186 @@ def complete_external_upload
     end
   end
 
+  def create_multipart
+    return render_404 if !SiteSetting.enable_direct_s3_uploads
+
+    RateLimiter.new(
+      current_user, "create-multipart-upload", CREATE_MULTIPART_RATE_LIMIT_PER_MINUTE, 1.minute
+    ).performed!
+
+    file_name = params.require(:file_name)
+    file_size = params.require(:file_size).to_i
+    upload_type = params.require(:upload_type)
```

Comment:
Do we allow any values to be stored here? I'm wondering if we should err on the safe side and only allow a fixed list of upload_types.

Reply:
We allow any value here and in the other `UploadController#create` action. I think we should validate the type in `UploadCreator`; I've put a TODO there and I will do that in another PR (I am not sure how that will affect plugins just yet).